In [1]:
import pandas as pd
import numpy as np

In [2]:
url = 'https://raw.githubusercontent.com/Yorko/mlcourse.ai/master/data/titanic_train.csv'
data = pd.read_csv(url)
data.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

Create a new feature age_category:

In [3]:
def age_category(age):
    """
    Parameters: 
    ----------
    age : int
        age of a passenger on board the Titanic.

    Returns
    -------
    1 for passengers whose less than 30 years old
    2 for passengers between 30 and 60 years old
    3 otherwise

    """
    if age < 30:
        return 1
    elif age <= 60:
        return 2
    elif age > 60:
        return 3

In [4]:
data["Age_category"] = data["Age"].apply(age_category)

data.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  Age_category  
0      0         A/5 21171   7.2500   NaN        S           1.0  
1      0          PC 17599  71.2833   C85        C           2.0  
2      0  STON/O2. 3101282   7.9250   NaN        S           1.0  
3      0            113803  53.1000  C123        S           2.0  
4      0            373450   8.0500   NaN        S           2.0

**Question 1:**
How many men/women were there onboard?

In [5]:
data["Sex"].value_counts()

male      577
female    314
Name: Sex, dtype: int64

**Question 2:**
Print the distribution of the Pclass feature.

In [6]:
data["Pclass"].value_counts()

3    491
1    216
2    184
Name: Pclass, dtype: int64

What is the distribution of Pclass for men and women separately. How many men from second class were there onboard?

In [7]:
data.groupby(["Sex"])["Pclass"].value_counts()

Sex     Pclass
female  3         144
        1          94
        2          76
male    3         347
        1         122
        2         108
Name: Pclass, dtype: int64

108 men from second class were on board.

**Question 3:**
What are median and standard deviation of Fare?. Round to two decimals.

In [9]:
print("The median fare is {} with a standard deviation of {}.".format(
    round(data["Fare"].median(), 2), round(data["Fare"].std(), 2)))

The median fare is 14.45 with a standard deviation of 49.69.


**Question 4:**
Is that true that the mean age of survived people is higher than that of passengers who eventually died?

In [10]:
print("Mean age of survived passenger", 
      round(data[data["Survived"] == 1]["Age"].mean(), 2))

Mean age of survived passenger 28.34


In [11]:
print("Mean age of those that died", 
      round(data[data["Survived"] == 0]["Age"].mean(), 2))

Mean age of those that died 30.63


False.

**Question 5:**
Is that true that passengers younger than 30 y.o. survived more frequently than those older than 60 y.o.? What are shares of survived people among young and old people?


In [12]:
data[data["Age_category"] == 1]["Survived"].value_counts(normalize = True)

0    0.59375
1    0.40625
Name: Survived, dtype: float64

In [13]:
data[data["Age_category"] == 3]["Survived"].value_counts(normalize = True)

0    0.772727
1    0.227273
Name: Survived, dtype: float64

True.

**Question 6:**
Is that true that women survived more frequently than men? What are shares of survived people among men and women?

In [14]:
data[data["Sex"] == "male"]["Survived"].value_counts(normalize = True)

0    0.811092
1    0.188908
Name: Survived, dtype: float64

In [15]:
data[data["Sex"] == "female"]["Survived"].value_counts(normalize = True)

1    0.742038
0    0.257962
Name: Survived, dtype: float64

True.

**Question 7:**
What's the most popular first name among male passengers?

In [16]:
data["First_name"] = data["Name"].apply(lambda name:name.split(",")[1].split()[1])

In [17]:
data[data["Sex"] == "male"]["First_name"].value_counts()

William      35
John         25
George       14
Charles      13
Thomas       13
             ..
Thor          1
Adolf         1
Milan         1
Francisco     1
Einar         1
Name: First_name, Length: 287, dtype: int64

**Question 8:**
How is average age for men/women dependent on Pclass? 

In [18]:
pd.crosstab(data["Sex"], data["Pclass"], values = data["Age"], aggfunc = np.mean)

Pclass          1          2          3
Sex                                    
female  34.611765  28.722973  21.750000
male    41.281386  30.740707  26.507589

* On average, men of 1 class are older than 40
* Men of all classes are on average older than women of the same class
* On average, passengers ofthe first class are older than those of the 2nd class who are older than passengers of the 3rd class